src/composite/tag/cache.jl

```diff
@@ -95,6 +95,7 @@ function apply!(r::AbstractRegister, c::CachedBlock, signal)
     return r
 end
 
+apply!(r::AbstractRegister, c::CachedBlock) = apply!(r, c.content)
```

Comment:
Why add this fallback?

maybe just error if it is not a defined register? since cache block is mainly for simulations and for `ArrayReg`

Reply:
If `CachedBlock` only caches `Array`, then it is natural to be ignored by other register types. When applying on a real device, I wish people do not have to change the circuit structure. Anyway, I am ok with removing it.

Reply:
I see. Maybe just keep it for convenience then.